Mathlib/Logic/Embedding/Basic.lean

```diff
@@ -0,0 +1,482 @@
+
+/-- The identity map as a `function.embedding`. -/
+@[refl, simps (config := { simpRhs := true })]
+protected def refl (α : Sort _) : α ↪ α :=
+  ⟨id, injective_id⟩
+#align function.embedding.refl Function.Embedding.refl
+
+-- porting note: in Lean 3 this was tagged `@[trans]`
+/-- Composition of `f : α ↪ β` and `g : β ↪ γ`. -/
+@[simps (config := { simpRhs := true })]
+protected def trans {α β γ} (f : α ↪ β) (g : β ↪ γ) : α ↪ γ :=
+  ⟨g ∘ f, g.injective.comp f.injective⟩
+#align function.embedding.trans Function.Embedding.trans
+
+instance : Trans Embedding Embedding Embedding := ⟨Embedding.trans⟩
+
+@[simp]
+theorem equiv_toEmbedding_trans_symm_toEmbedding {α β : Sort _} (e : α ≃ β) :
+    e.toEmbedding.trans e.symm.toEmbedding = Embedding.refl _ := by
+  ext
+  simp
+#align function.embedding.equiv_to_embedding_trans_symm_to_embedding
+  Function.Embedding.equiv_toEmbedding_trans_symm_toEmbedding
+
+@[simp]
+theorem equiv_symm_toEmbedding_trans_toEmbedding {α β : Sort _} (e : α ≃ β) :
+    e.symm.toEmbedding.trans e.toEmbedding = Embedding.refl _ := by
+  ext
+  simp
+#align function.embedding.equiv_symm_to_embedding_trans_to_embedding
+  Function.Embedding.equiv_symm_toEmbedding_trans_toEmbedding
+
+/-- Transfer an embedding along a pair of equivalences. -/
+@[simps (config := { fullyApplied := false, simpRhs := true })]
+protected def congr {α : Sort u} {β : Sort v} {γ : Sort w} {δ : Sort x} (e₁ : α ≃ β) (e₂ : γ ≃ δ)
+    (f : α ↪ γ) : β ↪ δ :=
+  (Equiv.toEmbedding e₁.symm).trans (f.trans e₂.toEmbedding)
+#align function.embedding.congr Function.Embedding.congr
+
+/-- A right inverse `surj_inv` of a surjective function as an `embedding`. -/
+protected noncomputable def ofSurjective {α β} (f : β → α) (hf : Surjective f) : α ↪ β :=
+  ⟨surjInv hf, injective_surjInv _⟩
+#align function.embedding.of_surjective Function.Embedding.ofSurjective
+
+/-- Convert a surjective `embedding` to an `equiv` -/
+protected noncomputable def equivOfSurjective {α β} (f : α ↪ β) (hf : Surjective f) : α ≃ β :=
+  Equiv.ofBijective f ⟨f.injective, hf⟩
+#align function.embedding.equiv_of_surjective Function.Embedding.equivOfSurjective
+
+/-- There is always an embedding from an empty type. -/
+protected def ofIsEmpty {α β} [IsEmpty α] : α ↪ β :=
+  ⟨isEmptyElim, isEmptyElim⟩
+#align function.embedding.of_is_empty Function.Embedding.ofIsEmpty
+
+/-- Change the value of an embedding `f` at one point. If the prescribed image
+is already occupied by some `f a'`, then swap the values at these two points. -/
+def setValue {α β} (f : α ↪ β) (a : α) (b : β) [∀ a', Decidable (a' = a)]
+    [∀ a', Decidable (f a' = b)] : α ↪ β :=
+  ⟨fun a' => if a' = a then b else if f a' = b then f a else f a', by
+    intro x y (h : ite _ _ _ = ite _ _ _)
+    -- This used to be almost automatic with `split_ifs` and `cc`
+    -- split_ifs  at h <;> try subst b <;> try simp only [f.injective.eq_iff] at * <;> cc⟩
+    simp only [ite_eq_iff, eq_ite_iff, and_true, f.injective.eq_iff] at h
```

Comment:
Could someone open an issue with a MWE of `split_ifs` behaving differently, so we can track this?